internal/controller/postgrescluster/pgbackrest.go

```diff
@@ -2111,7 +2112,7 @@ func (r *Reconciler) reconcileReplicaCreateBackup(ctx context.Context,
 		//   made in the spec (specifically a change to the config for an external repo).  Delete
 		//   and recreate the Job with proper hash per the current config.
 		if failed || replicaCreateRepoChanged ||
-			(job.GetAnnotations()[naming.PGBackRestCurrentConfig] != configName) ||
+			(job.GetAnnotations()[naming.PGBackRestCurrentConfig] != containerName) ||
```

Comment:
To double-check: in the old version, we were checking that this didn't equal either `pgbackrest.CMInstanceKey` or `pgbackrest.CMRepoKey`, but now it looks like we're checking that this isn't equal to either `naming.PGBackRestRepoContainerName` or `naming.ContainerDatabase` -- which works because we're setting that `annotation[naming.PGBackRestCurrentConfig]` to equal one of those container names.

It looks like that annotation label is only ever set and checked in this file, but is there a chance it's being used elsewhere?

Reply:
Right. This line is paired with L2153.

These two lines (and their tests) are the only place the annotation is used.

ConfigHash is also used on restore jobs. 🌱 Probably these two annotations/ideas need to be combined.